spec/basket_spec.rb

```diff
@@ -2,19 +2,60 @@
 
 RSpec.describe Basket do
 
-  it "creates basket" do
-    expect Basket.new.is_a? Basket
+  let(:football) { Product.new(name: "Football", price: 79) }
+  let(:basketball) { Product.new(name: "Basketball", price: 60) }
+  let(:tennis_ball) { Product.new(name: "Tennis ball", price: 10) }
+  let(:footballs_in_basket) { BasketItem.new(product: football, quantity: 2)}
+  let(:basketballs_in_basket) { BasketItem.new(product: basketball, quantity: 1)}
+  let(:tennis_balls_in_basket) { BasketItem.new(product: tennis_ball, quantity: 3)}
+  let(:basket) { Basket.new([footballs_in_basket, basketballs_in_basket]) }
+
+  describe "#sum" do
+    it "calculates products sum correctly" do
+      expect(basket.sum).to eq(218)
+    end
+  end
+
+  describe "#sum_with_vat" do
+    it "checks if sum with vat is correct" do
+      expect(basket.sum_with_vat).to eq(268.14)
+    end
   end
 
-  let(:p1) { Product.new(name: "Football", price: 79) }
-  let(:p2) { Product.new(name: "Basketball", price: 60) }
-  it "checks if sum with vat is correct" do
-    sum = Basket.new([p1, p2]).sum
-    sum_with_vat = Basket.new([p1, p2]).sum_with_vat
-    expect(sum * 1.23).to eq sum_with_vat
+  describe "#add" do
+    it "adds selected item to basket" do
+      basket.add(tennis_balls_in_basket)
+        products_in_basket_names = basket.items.map do |item|
+         { name: item.product.name }
+        end
+      expect(products_in_basket_names.select { |prod| prod[:name] == "Tennis ball" }.empty?).to eq(false)
```

Comment:
Will do. Just to know for future: is this for readability?
If yes, is there any rule of thumb for when to do it?
Or just common sense? :)